### PR TITLE
executor: set correct collate in show create table

### PR DESCRIPTION
**Pull request by jackysp**
Fix https://github.com/pingcap/docs-cn/issues/537

### CODE REVIEW

**Comment by jackysp** [comment]
PTAL @coocood @shenli @winkyao

File: executor/show_test.go [hunk 1]
```diff
@@ -296,6 +296,26 @@ func (s *testSuite) TestShow(c *C) {
 			"  `e` bigint(20) DEFAULT NULL\n" +
 			") ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin",
 	}
+	for i, r := range row {
+		c.Check(r, Equals, expectedRow[i])
+	}
+
+	// The following two cases are for https://github.com/pingcap/docs-cn/issues/537
```
Comment (coocood) [inline executor/show_test.go]: We better not reference a issue in `docs-cn`.

Comment (shenli) [reply executor/show_test.go]: Agree

File: executor/show.go [hunk 2]
```diff
@@ -457,6 +457,15 @@ func (e *ShowExec) fetchShowStatus() error {
 	return nil
 }
 
+func setDefaultCollate(charsetName string) string {
```
Comment (coocood) [inline executor/show.go]: s/setDefaultCollate/getDefaultCollate

File: executor/show_test.go [hunk 3]
```diff
@@ -296,6 +296,26 @@ func (s *testSuite) TestShow(c *C) {
...
+			"  `a` int(11) DEFAULT NULL\n"+
+			") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin",
+	))
+
+	tk.MustExec(`drop table if exists t`)
+	tk.MustExec(`create table t (a int) default charset=abcdefg`)
+	tk.MustQuery(`show create table t`).Check(testutil.RowsWithSep("|",
+		"t CREATE TABLE `t` (\n"+
+			"  `a` int(11) DEFAULT NULL\n"+
+			") ENGINE=InnoDB DEFAULT CHARSET=abcdefg",
```
Comment (shenli) [inline executor/show_test.go]: This is an invalid charset? Will MySQL accept it?

Comment (coocood) [reply executor/show_test.go]: It's an invalid charset, MySQL returns `Unknown character set` error.
But it is a different issue.

File: executor/show_test.go (hunk 1 above)
Comment (jackysp) [reply executor/show_test.go]: Done.

File: executor/show.go (hunk 2 above)
Comment (jackysp) [reply executor/show.go]: Done.

**Comment by jackysp** [comment]
PTAL @coocood

**Comment by coocood** [comment]
LGTM

**Pull request merged**